### PR TITLE
feat(iota): iota rebased poc

### PR DESCRIPTION
**Pull request by antbern**
## Motivation and Context

Explain why this change is necessary or the issue it addresses.

## Summary

Provide a short summary of changes in this pull request.

- Change 1
- Change 2

## Checklist

Please ensure that your PR meets the following requirements:

### General

- [ ] Code follows project style and guidelines.
- [ ] No redundant or duplicate code.
- [ ] No added new dependencies without clear justification and approval.

### Documentation

- [ ] Added/updated relevant documentation (e.g., README, inline comments).
- [ ] Added a CHANGELOG entry for major changes.

### Testing

- [ ] Added relevant test cases, leveraging tools like `rstest` or similar for reusable tests.
- [ ] Tests pass locally.

### Build & CI/CD

- [ ] Confirmed no CI/CD pipeline issues.
- [ ] Updated build scripts where necessary.

## Comments to Reviewer

Please provide any comments or points of consideration for the reviewers, such as specific areas of focus, potential edge cases, or context that would aid the review process.

### Reviewer Responsibilities

- [ ] Code adheres to style and guidelines.
- [ ] All tests are appropriately covered and they pass.
- [ ] Consider performance and/or security impacts.
- [ ] Constructive feedback is provided.

### CODE REVIEW

File: sdk/Cargo.toml [hunk 1]
```diff
@@ -70,13 +60,25 @@ iota-sdk = { package = "etospheres-labs-iota-sdk", version = "1.1.5", default-fe
     # these are enabled in the "default" feature
     # "jammdb",
 ] }
+
+# crates needed for iota-rebased integration, should be compatible versions
+# iota-sdk-rebased = { package = "iota-sdk", git = "https://github.com/iotaledger/iota/", branch = "develop", default-features = false, features = []}
+# bip32 = "0.5"
+# anyhow = "1"
+# signature = "1.6.0"
```
Comment (hauju) [inline sdk/Cargo.toml:68]: Wanna keep for later?

File: sdk/examples/13_send_amount_rebased.rs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    sdk.create_wallet_from_existing_mnemonic(&user.pin, &user.mnemonic)
+        .await
+        .unwrap();
+
+    // Set network
+    sdk.set_networks(vec![ApiNetwork {
+        key: "iota-rebased".to_string(),
+        is_testnet: true,
+        display_name: "IOTA Rebased".to_string(),
+        display_symbol: "IOTAR".to_string(),
```
Comment (hauju) [inline sdk/examples/13_send_amount_rebased.rs:36]: The Symbol should be just IOTA

File: sdk/examples/13_send_amount_rebased.rs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        block_explorer_url: String::new(),
+    }]);
+    sdk.set_network("iota-rebased".to_string()).await.unwrap();
+
+    // Generate new address
+    let recipient_address = sdk.generate_new_address(&user.pin).await.unwrap();
+    let balance = sdk.get_balance(&user.pin).await.unwrap();
+    println!("address: {recipient_address}, balance: {balance:?}");
+
+    // let recipient_address = "0x393bdb466e8efc29ca5905f23d90b069d2b00f35016dd316039c97b0363ce217".to_string();
```
Comment (hauju) [inline sdk/examples/13_send_amount_rebased.rs:55]: Remove old comments

File: sdk/examples/13_send_amount_rebased.rs [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+    // Send amount
+    let amount = dec!(1.0).try_into().unwrap();
+    let data = Some("test".to_string().into_bytes());
+    // estimate gas
+    // let estimate = sdk
```
Comment (hauju) [inline sdk/examples/13_send_amount_rebased.rs:61]: Can be removed?

**Review by hauju — COMMENTED** [review]
Some small comments.

File: sdk/examples/13_send_amount_rebased.rs (hunk 4 above)
Comment (antbern) [reply sdk/examples/13_send_amount_rebased.rs:61]: I would keep it for now since we should uncomment this once we have gas estimation implemented 🚀

File: sdk/examples/13_send_amount_rebased.rs (hunk 2 above)
Comment (antbern) [reply sdk/examples/13_send_amount_rebased.rs]: fixed!

File: sdk/Cargo.toml (hunk 1 above)
Comment (antbern) [reply sdk/Cargo.toml]: Oops, removed!

File: sdk/examples/13_send_amount_rebased.rs (hunk 3 above)
Comment (antbern) [reply sdk/examples/13_send_amount_rebased.rs]: Removed ✔️

**Review by hauju — APPROVED** [review]

**Review by lszetg — APPROVED** [review]
a lot of work :D great that you left comments, it was much easier to understand what’s going on :D hats off 🎉

File: sdk/src/wallet/rebased/types/message_envelope.rs [hunk 5]
```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Mysten Labs, Inc.
+// Modifications Copyright (c) 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+//
```
[the remark this answers is not in the document]
Comment (antbern) [reply sdk/src/wallet/rebased/types/message_envelope.rs:4]: Good catch, I will add it!

**Review by lszetg — APPROVED** [review]
